src/messages/manifestjson.js

```diff
@@ -35,6 +35,16 @@ export const MANIFEST_FIELD_INVALID = {
   file: MANIFEST_JSON,
 };
 
+export const MANIFEST_BAD_PERMISSION = {
+  code: 'MANIFEST_BAD_PERMISSION',
+  legacyCode: null,
+  message: _('The permission type is unsupported.'),
+  description: _(singleLineString`Permissions must be strings; object
```

Comment:
Ah ok - makes sense re the app thing. Are there any other indicators of an App versus ext? This still seems worth adding but I wonder if there's a more solid thing that we could also be checking for?